aspnet/mvc/controllers/routing.rst

```diff
@@ -1,8 +1,140 @@
+
+.. code-block:: c#
+
+  [Route("category/{movie}.rss", Order = 0, Name = "MovieCategoryFeed")]
+
+A common scenario for using attribute routing in an API controller is having the same route behave differently depending upon which HTTP verb is used. For example, you might want a page that displays a movie by its id using the route ``/movies/1337``, but also on that page be able to purchase the movie, and send the results back to the server. To restrict the HTTP verb that is accepted, apply the ``[HttpGet]`` attribute to the ``GetMovie`` action:
+
+.. code-block:: c#
+
+  [HttpGet("{id}", Name = "GetMovie")]
+  public IActionResult GetMovie([FromRoute] int id)
+
+The order in which routes are processed is important, but you can control how it works. First, routing inspects the ``RouteOrder`` property of the ``[Route]`` attribute. 
+
+Routes on controller vs actions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Routes map URLs to actions in controllers. When at the action level, constraints may be applied to the actions to further restrict activity. 
+
+Inheritance
+^^^^^^^^^^^
+
+The ``IRouteConstraint`` interface exposes a ``Match`` method that you can use to create your own route constraints via inheritance, as the code below demonstrates:
+
+.. code-block:: c#
+
+  public class RouteConstraint : IRouteConstraint
+  {
+    public bool Match(HttpContext httpContext, 
+        IRouter route, string routeKey, 
+        IDictionary<string, object> values, 
+        RouteDirection routeDirection)
+        {
+          // code to build constraint
+        }
+  }
+
+Notice that the ``Match`` method contains parameters that you can use to inspect the route and HTTP request.
+  
+Token replacement in route templates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since maintainability is important, you can use token replacements in route templates. Tokens are substitutions for actual route values at runtime, and you may define them during startup for conventional routes, or in attributes for attribute routes. You may use regular expressions to create route tokens. Consider the following route attribute:
+
+.. code-block:: c#
+
+  [Route("category/{movie}.rss", Order = 0, Name = "MovieCategoryFeed")]
+  public IActionResult MovieRSS(string name)
+
+The above creates a route that directs to the ``MovieRSS`` method, and looks something like the following:
+
+.. code-block:: c#
+
+  /category/GoneWithTheWind.rss 
+
+You may create custom route templates if you can't find any other built-in solution. Generally, it is because you need to dynamically compute a route template. To do so, subclass any route attribute, or implement the ``IApplicationModelConvention`` interface. Keep the code in the custom attribute simple as a limited extensibility point.   
```

Comment:
You create custom route templates when the built in or default routes won’t work (something like that).
What’s a built in route? What’s a default route  - maybe we need to define them. Need an example where you need to create one.